File: easy/BinaryNumWithAlternateBits.js
// Given a positive integer, check whether it has alternating bits: namely, if two adjacent bits will always have different values.

const hasAlternatingBits = function(n) {
  const bin = (n >>> 0).toString(2);
  for (let i = 1; i < bin.length; i++) {
      if (bin[i] === bin[i - 1]) {
          return false;
      }
  }
  return true;
};

console.log(hasAlternatingBits(5) === true)

File: easy/BinaryNumWithAlternateBits.py
# Given a positive integer, check whether it has alternating bits: namely, if two adjacent bits will always have different values.

def hasAlternatingBits(n):
    bStr = bin(n)
    i = 3
    while i < len(bStr):
        if bStr[i] == bStr[i - 1]:
            return False
        i += 1
    return True


print(hasAlternatingBits(5) == True)
print(hasAlternatingBits(7) == False)
